tests/test_git_browse_ci.py

```diff
@@ -0,0 +1,174 @@
+import os, subprocess
+from testpath import MockCommand, modified_env
+
+github_origin = "https://github.com/tj/git-extras"
+gitlab_origin = "https://gitlab.com/tj/git-extras"
+bitbucket_origin = "https://bitbucket.org/tj/git-extras"
+unknown_site_origin = "https://unknown-site.com/tj/git-extras"
+
+def set_origin_url(git, url):
+    git.remote("set-url", "origin", url + ".git")
+
+def create_expected_action_url(git, mode):
+    if mode == "github":
+        return github_origin + '/actions'
+    if mode == "gitlab":
+        return gitlab_origin + '/-/pipelines'
+    if mode == "bitbucket":
+        return bitbucket_origin + '/addon/pipelines/home'
+
+class TestGitBrowse:
+    def test_browse_github_ci_on_mac(self, temp_repo):
+        git = temp_repo.get_repo_git()
+        git.remote("add", "origin", github_origin + ".git")
```

Comment:
I'm having trouble figuring out why some tests use `set_origin_url` and others don't. I thought a new Git repository was created for each test, so "git remote add" should be done for all of them?

Reply:
It should be all used with `set_origin_url` or a code error.

Reply:
I got your meaning, emmm, 

> I thought a new Git repository was created for each test
Nope, a test repository for testing one command

> so "git remote add" should be done for all of them?
One temp repository created for testing one command , so initialize it once and then use it again.

Reply:
It is my bad without any document to describe it, I will fill it later with some refactors.